src/compiler/GF/Compile/Compute/Predef.hs

```diff
@@ -27,6 +27,7 @@ instance Predef Int where
 
 instance Predef Bool where
   toValue = boolV
+  fromValue boolV = return (boolV == boolV)
```

Comment:
This looks strange. If `boolV` is False, then you return `False==False` i.e. `True`. What do the other definitions do? If unsure, you could also just leave this undefined

Reply:
The value is of the datatype [Value](https://github.com/ffrixslee/gf-core/blob/99b04adc73a513609700dba96b90d007c23d85bc/src/compiler/GF/Compile/Compute/Value.hs#L9-L38), defined in Value.hs.

If you look at fromValue for Int or String, you see how it matches with the constructors [`VInt` and `VString`](https://github.com/ffrixslee/gf-core/blob/99b04adc73a513609700dba96b90d007c23d85bc/src/compiler/GF/Compile/Compute/Value.hs#L17-L19). But there's no `VBool`: in fact, toValue for Bool returns something constructed by [`VCApp`](https://github.com/ffrixslee/gf-core/blob/99b04adc73a513609700dba96b90d007c23d85bc/src/compiler/GF/Compile/Compute/Value.hs#L11).

So I suggest leaving `fromValue` for Bool unimplemented.